livox_ros2_driver/livox_ros2_driver/lddc.cpp

```diff
@@ -718,427 +709,461 @@ void Lddc::onDiagnosticsTimer()
   for (uint32_t i = 0; i < kMaxSourceLidar; ++i) {
     if (lds_->lidars_[i].handle != kMaxSourceLidar) ++lidar_count_;
   }
+
+  if (connected_lidar_count_ != lds_->connected_lidars_.size())
+  {
+    connected_lidar_count_ = lds_->connected_lidars_.size();
+    for (const auto &lidar : lds_->connected_lidars_)
+    {
+      std::string broadcast_code = lidar.first;
+
+      updater_.add(
+          fmt::format("livox_temperature_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkTemperature, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_internal_voltage_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkVoltage, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_motor_status_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkMotor, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_optical_window_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkDirty, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_firmware_status_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkFirmware, this, std::placeholders::_1,
+              broadcast_code));
+
+      if (check_pps_signal_)
+      {
+        updater_.add(
+            fmt::format("livox_pps_signal_{}", broadcast_code),
+            std::bind(
+                &Lddc::checkPPSSignal, this, std::placeholders::_1,
+                broadcast_code));
+      }
+
+      updater_.add(
+          fmt::format("livox_service_life_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkServiceLife, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_fan_status_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkFan, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_time_sync{}", broadcast_code),
+          std::bind(
+              &Lddc::checkTimeSync, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_connection_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkConnection, this, std::placeholders::_1,
+              broadcast_code));
+    }
+  }
+
   updater_.force_update();
 }
 
-void Lddc::checkTemperature(diagnostic_updater::DiagnosticStatusWrapper & stat)
+void Lddc::getLidarByBroadcastcode(std::pair<std::string, DeviceInfo *> &lidar,
+                                   const std::string &broadcast_code)
 {
-  if (lidar_count_ == 0) {
-    stat.summary(DiagStatus::WARN, "No LiDARs Connected");
-    return;
+  for (const auto & tmp_lidar : lds_->connected_lidars_) {
```

Comment:
I think `tmp_lidar` -> `connected_lidar` is better.

Reply:
I changed it.